mobile-widgets/qmlmanager.cpp

```diff
@@ -1375,6 +1375,15 @@ bool QMLManager::undoDelete(int id)
 	return true;
 }
 
+void QMLManager::selectDive(int id)
```

Comment:
`amount_selected` doesn't seem to be used on mobile. Nevertheless, I'd suggest that this function keeps it up to date to avoid surprises.

Something like:
```
	amount_selected = 0;
	for_each_dive (i, dive) {
		dive->selected = (dive->id == id);
		if (dive->selected)
			++amount_selected;
	}
```

I can't comment on the rest of the PR, since I don't understand it.

BTW: Would it be possible to change the details-view so that only one DiveObjectHelper object is generated per page load? I don't understand that property thing.

Reply:
I think you are exaggerating if you say you  don't "understand" the rest of the PR. It's really not that hard - just awkward and backwards.

As to your code snippet. That seems a somewhat convoluted way to set `amount_selected = 1` since we sure hope that those `id`s are unique and only one is passed in, right?

And limiting the number of `DiveObjectHelper`s generated per page load... right now I'm mostly aiming for "have this stupid thing work at all". I agree that it would nice to have it also be efficient, but one miracle at a time, please.

Reply:
> As to your code snippet. That seems a somewhat convoluted way to set amount_selected = 1 since we sure hope that those ids are unique and only one is passed in, right?

Unless an unknown id is passed in. Shouldn't happen, but who knows...

Reply:
For the sake of robustness you could even add a
```
	if (amount_selected == 0)
		qWarning("QManager::selectDive(): unknown id")
```
or similar at the end of the function.